fs_attachment/models/ir_attachment.py

```diff
@@ -374,13 +374,28 @@ def _set_attachment_data(self, asbytes) -> None:  # pylint: disable=missing-retu
     def _storage_file_read(self, fname: str) -> bytes | None:
         """Read the file from the filesystem storage"""
         fs, _storage, fname = self._fs_parse_store_fname(fname)
-        try:
-            with fs.open(fname, "rb") as f:
-                return f.read()
-        except IOError:
-            _logger.info(
-                "Error reading %s on storage %s", fname, _storage, exc_info=True
-            )
+        read_retry_attempts = (
```

Comment:
` _storage_file_read` is an `api.model` method. You should not access the storage via the `fs_strorage_id` field.
To get an instance of the `fs.storage` record you can use: 
```python
fs, storage_code, fname = self._fs_parse_store_fname(fname)
if storage_code:
    storage = self.env["fs.storage"].get_by_code(storage_code)

```

To avoid to always get access to the `fs.storage` record from the database, you could use a dedicated method decorated with `orm.cache`in the FSStorage class

```python
@api.model
@orm.cache("code")
def get_read_retry_config(self, code) -> tuple[float, float]:
    """ ...
    """
    storage = self.get_by_code(code)
    return max(storage.read_retry_attempts, 0), max(storage.read_retry_delay, 0.0)

Reply:
Thank you for help @lmignon I adapted the code to not rely on self inside api.model method. I did some testing and it worked, it only access get_read_retry_config on the first attempt